lib/std/os/windows.zig

```diff
@@ -818,14 +819,27 @@ pub fn DeleteFile(sub_path_w: []const u16, options: DeleteFileOptions) DeleteFil
         0,
     );
     switch (rc) {
-        .SUCCESS => return CloseHandle(tmp_handle),
+        .SUCCESS => CloseHandle(tmp_handle),
         .OBJECT_NAME_INVALID => unreachable,
         .OBJECT_NAME_NOT_FOUND => return error.FileNotFound,
         .INVALID_PARAMETER => unreachable,
         .FILE_IS_A_DIRECTORY => return error.IsDir,
         .NOT_A_DIRECTORY => return error.NotDir,
         else => return unexpectedStatus(rc),
     }
+
+    if (options.remove_dir){
```

Comment:
Would you mind adding a comment for posterity explaining why we need to test for the dir existence here?

Reply:
Added @kubkon